### PR TITLE
Tempus add ssp rk methods

### PR DESCRIPTION
**Pull request by sconde**
@trilinos/tempus

### CODE REVIEW

File: packages/tempus/src/Tempus_StepperFactory.hpp [hunk 1]
```diff
@@ -840,6 +858,27 @@ class StepperFactory
...
+  {
+    auto stepper = Teuchos::rcp(new StepperERK_2StageTheta<Scalar>());
+    setStepperRKValues(stepper, stepperPL);
+    if (stepperPL != Teuchos::null) {
+      stepper->setTheta(stepperPL->get<double>("theta", 0.5));
+    }
+
+    if (model != Teuchos::null) {
+      stepper->setModel(model);
+      //setStepperSolverValues(stepper, stepperPL);
```
Comment (ccober6) [inline packages/tempus/src/Tempus_StepperFactory.hpp]: Still need this?

File: packages/tempus/src/Tempus_StepperFactory.hpp [hunk 2]
```diff
@@ -955,6 +994,83 @@ class StepperFactory
...
 
+  Teuchos::RCP<StepperSDIRK_SSPDIRK22<Scalar> >
+  createStepperSDIRK_SSPDIRK22(
+    const Teuchos::RCP<const Thyra::ModelEvaluator<Scalar> >& model,
+    Teuchos::RCP<Teuchos::ParameterList> stepperPL)
+  {
+    auto stepper = Teuchos::rcp(new StepperSDIRK_SSPDIRK22<Scalar>());
+    setStepperDIRKValues(stepper, stepperPL);
+    //if (stepperPL != Teuchos::null)
+      //stepper->setGamma(stepperPL->get<double>("gamma", 0.2928932188134524));
```
Comment (ccober6) [inline packages/tempus/src/Tempus_StepperFactory.hpp]: Same here.

File: packages/tempus/src/Tempus_StepperFactory.hpp [hunk 3]
```diff
@@ -955,6 +994,83 @@ class StepperFactory
...
+
+  Teuchos::RCP<StepperSDIRK_SSPDIRK32<Scalar> >
+  createStepperSDIRK_SSPDIRK32(
+    const Teuchos::RCP<const Thyra::ModelEvaluator<Scalar> >& model,
+    Teuchos::RCP<Teuchos::ParameterList> stepperPL)
+  {
+    auto stepper = Teuchos::rcp(new StepperSDIRK_SSPDIRK32<Scalar>());
+    setStepperDIRKValues(stepper, stepperPL);
+    //if (stepperPL != Teuchos::null)
+      //stepper->setGamma(stepperPL->get<double>("gamma", 0.2928932188134524));
```
Comment (ccober6) [inline packages/tempus/src/Tempus_StepperFactory.hpp]: Same here.

File: packages/tempus/src/Tempus_StepperFactory.hpp [hunk 4]
```diff
@@ -955,6 +994,83 @@ class StepperFactory
...
+
+  Teuchos::RCP<StepperSDIRK_SSPDIRK23<Scalar> >
+  createStepperSDIRK_SSPDIRK23(
+    const Teuchos::RCP<const Thyra::ModelEvaluator<Scalar> >& model,
+    Teuchos::RCP<Teuchos::ParameterList> stepperPL)
+  {
+    auto stepper = Teuchos::rcp(new StepperSDIRK_SSPDIRK23<Scalar>());
+    setStepperDIRKValues(stepper, stepperPL);
+    //if (stepperPL != Teuchos::null)
+      //stepper->setGamma(stepperPL->get<double>("gamma", 0.2928932188134524));
```
Comment (ccober6) [inline packages/tempus/src/Tempus_StepperFactory.hpp]: Same here.

File: packages/tempus/src/Tempus_StepperFactory.hpp [hunk 5]
```diff
@@ -955,6 +994,83 @@ class StepperFactory
...
+
+  Teuchos::RCP<StepperSDIRK_SSPDIRK33<Scalar> >
+  createStepperSDIRK_SSPDIRK33(
+    const Teuchos::RCP<const Thyra::ModelEvaluator<Scalar> >& model,
+    Teuchos::RCP<Teuchos::ParameterList> stepperPL)
+  {
+    auto stepper = Teuchos::rcp(new StepperSDIRK_SSPDIRK33<Scalar>());
+    setStepperDIRKValues(stepper, stepperPL);
+    //if (stepperPL != Teuchos::null)
+      //stepper->setGamma(stepperPL->get<double>("gamma", 0.2928932188134524));
```
Comment (ccober6) [inline packages/tempus/src/Tempus_StepperFactory.hpp]: Same here.

File: packages/tempus/src/Tempus_StepperRKButcherTableau.hpp [hunk 6]
```diff
@@ -2344,6 +2597,452 @@ class StepperSDIRK_ImplicitMidpoint :
...
+ *  \begin{array}{c|c}
+ *    c & A \\ \hline
+ *      & b^T \\ \hline
+ *      & \hat{b}^T
+ *  \end{array}
+ *  \;\;\;\;\mbox{ where }\;\;\;\;
+ *  \begin{array}{c|cccc}  1/6  & 1/6  & \\
+ *                         1/2  & 1/3  & 1/6 & \\   \hline
+ *                         5/6  & 1/3  & 1/3 & 1/3 \\   \hline
+ *                              & 1/3  & 1/3 & 1/3  \end{array}
```
Comment (ccober6) [inline packages/tempus/src/Tempus_StepperRKButcherTableau.hpp]: This doxygen does not match tableau.

File: packages/tempus/test/ExplicitRK/Tempus_ExplicitRKTest.cpp [hunk 7]
```diff
@@ -37,10 +37,11 @@ using Tempus::SolutionHistory;
...
 
 // Comment out any of the following tests to exclude from build/run.
-#define TEST_PARAMETERLIST
-#define TEST_CONSTRUCTING_FROM_DEFAULTS
+//#define TEST_PARAMETERLIST
+//#define TEST_CONSTRUCTING_FROM_DEFAULTS
 #define TEST_SINCOS
-#define TEST_EMBEDDED_VANDERPOL
+//#define TEST_EMBEDDED_VANDERPOL
+//#define TEST_EMBEDDED_VANDERPOL_PID
```
Comment (ccober6) [inline packages/tempus/test/ExplicitRK/Tempus_ExplicitRKTest.cpp]: Need to uncomment and retest these.

File: packages/tempus/test/ExplicitRK/Tempus_ExplicitRKTest.cpp [hunk 8]
```diff
@@ -395,6 +406,12 @@ TEUCHOS_UNIT_TEST(ExplicitRK, SinCos)
 
     double order_tol = 0.01;
     if (RKMethods[m] == "Merson 4(5) Pair") order_tol = 0.04;
+    else if (RKMethods[m] == "SSPERK54")    order_tol = 0.06;
+
+
+    //asm("int $3");
+    //std::cout << "SIDAFA: got here!!" << std::endl;
```
Comment (ccober6) [inline packages/tempus/test/ExplicitRK/Tempus_ExplicitRKTest.cpp]: Can we remove this?

**Comment by ccober6** [comment]
Looks like great additions!

**Comment by sconde** [comment]
@ccober6 Can you approve this PR again please

**Pull request merged**